.github/workflows/test.yml

```diff
@@ -0,0 +1,108 @@
+name: test
+
+on:
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+
+jobs:
+  # Test the Android ds2 binary on an emulator.
+  android-test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        # We don't currently have a good way of running arm64-v8a tests on the
+        # available GitHub test runners.
+        abi: ['x86_64']
+
+    env:
+      ANDROID_API_LEVEL: 28
+      DEBUG_PORT: 5432
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: seanmiddleditch/gha-setup-ninja@master
```

Comment:
Can we pin to a revision instead? The branch name pinning is considered a supply chain attack vector.